py/farm_ng/core/events_file_reader.py

```diff
@@ -95,19 +113,19 @@ def close(self) -> bool:
         return self.is_closed()
 
     def _reset_event_index(self) -> None:
-        self._event_index = defaultdict(list)
+        self._events_index = []
 
-    def uris(self) -> List[Uri]:
-        if not self._event_index:
+    def get_uris(self) -> List[Uri]:
+        if len(self._events_index) == 0:
             self._build_event_index()
-        return [string_to_uri(key) for key in sorted(self._event_index.keys())]
 
-    def has_uri(self, uri: Uri) -> bool:
-        if self._event_index is None:
-            return False
-        return uri_to_string(uri) in self._event_index.keys()
+        uris_set: Set[str] = set()
```

Comment:
Ideally this is built once during the index constructor so that it's constant small time to call.